mmv1/templates/terraform/examples/forwarding_rule_service_directory_registration.tf.erb

```diff
@@ -0,0 +1,28 @@
+resource "google_compute_forwarding_rule" "<%= ctx[:primary_resource_id] %>" {
+  name       = "<%= ctx[:vars]['forwarding_rule_name'] %>"
+  target     = google_compute_target_pool.default.id
+  port_range = "80"
+  serviceDirectoryRegistrations = {
```

Comment:
You may need to use underscore between words instead of the capital letters.

Reply:
Thanks, will give that a try :)

Reply:
Reworked it with the underscores and removed the `=` and things started working. The resource successfully got created in the acceptance test this AM.

```
=== CONT  TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample
    vcr_utils.go:152: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # google_compute_forwarding_rule.default must be replaced
        -/+ resource "google_compute_forwarding_rule" "default" {
              - all_ports               = false -> null
              - allow_global_access     = false -> null
              - allow_psc_global_access = false -> null
              + base_forwarding_rule    = (known after apply)
              ~ creation_timestamp      = "2023-09-12T05:30:18.925-07:00" -> (known after apply)
              ~ id                      = "projects/landing-zone-ctrl/regions/us-central1/forwardingRules/default" -> (known after apply)
              ~ ip_address              = "35.193.254.109" -> (known after apply)
              ~ ip_protocol             = "TCP" -> (known after apply)
              + ip_version              = (known after apply)
              - is_mirroring_collector  = false -> null
              ~ label_fingerprint       = "42WmSpB8rSM=" -> (known after apply)
                name                    = "default"
              + network                 = (known after apply)
              ~ network_tier            = "PREMIUM" -> (known after apply)
              ~ port_range              = "80-80" -> "80"
              ~ project                 = "landing-zone-ctrl" -> (known after apply)
              + psc_connection_id       = (known after apply)
              + psc_connection_status   = (known after apply)
              ~ region                  = "us-central1" -> (known after apply)
              ~ self_link               = "https://www.googleapis.com/compute/v1/projects/landing-zone-ctrl/regions/us-central1/forwardingRules/default" -> (known after apply)
              + service_name            = (known after apply)
              + subnetwork              = (known after apply)
              ~ target                  = "https://www.googleapis.com/compute/beta/projects/landing-zone-ctrl/regions/us-central1/targetPools/tf-test-target-poolyd7cg7mghe" -> "projects/landing-zone-ctrl/regions/us-central1/targetPools/tf-test-target-poolyd7cg7mghe"
                # (1 unchanged attribute hidden)
        
              ~ service_directory_registrations {
                  + service_directory_region = "us-central1" # forces replacement
                    # (2 unchanged attributes hidden)
                }
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
        ```

Reply:
Tried adding it to `testAccComputeForwardingRule_serviceDirectoryRegistrations`  and it also fails

```
service_directory_registrations {
    namespace = google_service_directory_namespace.examplens.namespace_id
    service = google_service_directory_service.examplesvc.service_id
    service_directory_region = "us-central1"
  }
```

Resource creates as expected and appears in the UI.

Reply:
This test is supposed to create the resource and delete it in the end. If it fails in the middle the resource will not be removed automatically and the resource state will not be properly recorded.

Reply:
Ahh good to know. Rebuilding the tests and will run again with DEBUG enabled to hopefully find out where exactly it's breaking.

Reply:
Thanks and good luck!

Reply:
Appreciate the help and patience! First time through the magic modules process. 

Think I've narrowed it down it's failing to find the forwarding rule

```
{
  "error": {
    "code": 404,
    "message": "The resource 'projects/landing-zone-ctrl/regions/us-central1/forwardingRules/tf-test-mnvnjjhdk2' was not found",
    "errors": [
      {
        "message": "The resource 'projects/landing-zone-ctrl/regions/us-central1/forwardingRules/tf-test-mnvnjjhdk2' was not found",
        "domain": "global",
        "reason": "notFound"
      }
    ]
  }
}

```

Reply:
Could you upload the full log? We need to validate which step shows this error, because if we see it after the deletion that would be the expected behaviour.

Reply:
Will retrace mysteps, made some changes and it's all passing now.

```
make testacc TEST=./google-beta/services/compute TESTARGS='-run=TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample'
==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta/services/compute -v -run=TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample
=== PAUSE TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample
=== CONT  TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample
--- PASS: TestAccComputeForwardingRule_forwardingRuleServiceDirectoryRegistrationExample (54.07s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute        54.112s
```

Reply:
Healthy dose of skeptism says I broke things into being "fixed"

Reply:
Looks like there's still a diff for the new field. Could you take a look into your log and see in any of the GET request if `service_directory_region` shows up?